lib/recipes-data/src/lib/fastly.ts

```diff
@@ -0,0 +1,103 @@
+ * the next time a client asks for it.
+ * This function does not retry; use the `sendFastlyPurgeRequestWithRetries` function for that
+ *
+ * Note, the CONTENT_URL_BASE parameter must be configured (either in environment variables or via a mock when testing).
+ * The function will throw if it is not defined.
+ * @param contentPath URL to purge. This is relative to the configured CONTENT_URL_BASE
+ * @param apiKey Fastly API key to authenticate the request.  The function will throw if this is undefined or empty.
+ * @param purgeType Whether to execute a soft or hard purge (default soft). See the docs on PurgeType for more information.
+ */
+export async function sendFastlyPurgeRequest(contentPath:string, apiKey:string, purgeType?: PurgeType) {
+  if(!ContentPrefix) throw new Error("Cannot purge because CONTENT_URL_BASE is not set");
+  if(!apiKey || apiKey=="") throw new Error("Cannot purge because Fastly API key is not set");
+
+  const urlToPurge = [
+    "https://api.fastly.com/purge",
+    removeLeadingAndTrailingSlash(ContentPrefix),
+    removeLeadingAndTrailingSlash(contentPath)
+  ].join("/");
+
+  const baseHeaders:Record<string,string> = {
+    "FASTLY-KEY": apiKey,
+    "Accept": "application/json",
+  };
+
+  const headers = purgeType=="hard" ? {
+    "Fastly-Soft-Purge": "1",
+    ...baseHeaders
+  } : baseHeaders;
+
+  if(DebugLogsEnabled) console.debug("urlToPurge is ", urlToPurge);
+  const response = await fetch(urlToPurge, {
+    method: "POST",
+    headers,
+  });
+  const content = await response.text();
+
+  switch(response.status) {
+    case 200:
+      if(DebugLogsEnabled) console.log(`Purge of ${contentPath} successful: ${content}`);
+      break;
+    default:
+      console.error(`Unable to purge ${contentPath}, Fastly returned ${response.status}: ${content}`);
+      throw new FastlyError(`Fastly returned ${response.status}`);
+  }
+}
+
+/**
+ * calls `sendFastlyPurgeRequest` with the given parameters.  If Fastly returns an error, this will delay by the
+ * number of seconds given in RETRY_DELAY and then retry, up to a maximum of MAX_RETRIES attempts.
+ * @param contentPath URL to purge. This is relative to the configured CONTENT_URL_BASE
+ * @param apiKey Fastly API key to authenticate the request.  The function will throw if this is undefined or empty.
+ * @param purgeType Whether to execute a soft or hard purge (default soft). See the docs on PurgeType for more information.
+ * @param retryCount don't specify this, it's used internally.
+ */
+export async function sendFastlyPurgeRequestWithRetries(contentPath:string, apiKey:string, purgeType?: PurgeType, retryCount?:number):Promise<void>
+{
+  try {
+    return sendFastlyPurgeRequest(contentPath, apiKey, purgeType);
+  } catch(err) {
+    if(err instanceof FastlyError) {
+      const nextRetry = retryCount ? retryCount + 1 : 1;
+      if(nextRetry>MaximumRetries) throw err; //we give up! it ain't gonna work.
```

Comment:
I know `MaximumRetries` has a default in config, what happens if that config is removed? Worth adding a guard here to prevent infinite retries? Maybe it'll collapse with an error anyway if it's `undefined`?

Reply:
Good point i'll have a look